geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKeyCommandsFunctionExecutor.java

```diff
@@ -93,7 +94,61 @@ public String internalType(RedisKey key) {
 
   @Override
   public boolean rename(RedisKey oldKey, RedisKey newKey) {
-    // caller has already done all the stripedExecutor locking
-    return getRedisData(oldKey).rename(getRegion(), oldKey, newKey);
+    RedisKey firstLock;
+    RedisKey secondLock;
+    if (arrayCmp(oldKey.toBytes(), newKey.toBytes()) > 0) {
```

Comment:
I noticed we're just comparing the bytes now.... I'm wondering if we'll have issues if both keys are on the same stripe.  We might encounter a deadlock.  I think that would eliminate the need for the byte array comparison.

Reply:
Should be fixed now.